src/aggregate-release-notes.js

```diff
@@ -86,92 +111,118 @@ function getUpdatedDependencies(prevVer, currentVer) {
   return updatedDependencies;
 }
 
+/*
+ * diff :: {currentVer: '3.1.4', prevVer: '3.1.2'}
+ * Returns [{name: 'v3.1.3', zipball_url, commit, .., ...]
+ */
 async function matchTags(repo, diff, { provider, owner }) {
+  let res;
   try {
-    //the maximum number of match tags to return
-    const upperBound = 10;
-    let tags = [];
-    const res = await provider.listTags(owner, repo);
-
-    for (let i = 0; i < res.data.length; i++) {
-      let currentRef = res.data[i].name.replace(/^v/, '');
-
-      if (diff.prevVer && currentRef === diff.prevVer || tags.length >= upperBound) {
-        break;
-      }
-
-      tags.push(res.data[i]);
-    }
-    // sort in ascending order by ref
-    return tags.sort((v1, v2) => semver.gt(v1.name, v2.name));
+    res = await provider.listTags(owner, repo, {page1: 6});
   } catch(err) {
     console.error('Error in matchTags', err);
   }
+  return filterTags(res.data, diff);
+}
+
+function filterTags(allTags, diff) {
+  //the maximum number of match tags to return
+  const upperBound = 10;
+
+  // We keep prevVer (>=) in the list to be able to get `previousRelease` and `previousReleaseSha` in the next step.
+  let tags = allTags.filter(tag => (
+    semver.satisfies(tag.name, `>=${diff.prevVer} <=${diff.currentVer}`)
+  ));
+
+  // sort in ascending order by ref
+  return tags
+    .filter((t, i) => i <= upperBound)
+    .sort((v1, v2) => semver.gt(v1.name, v2.name));
 }
 
-async function getAllReleaseNotes(updatedDependencies, options) {
-  const { owner, provider } = options;
-  const matchingTags = [];
+async function getAllReleaseNotes(updatedDependencies, { owner, repo, provider }) {
+  const matchingTags = {};
   let releaseNotes = {};
   for (let key in updatedDependencies) {
     try {
-      matchingTags[key] = await matchTags(key, updatedDependencies[key], options);
-
+      matchingTags[key] = await matchTags(key, updatedDependencies[key], { owner, repo, provider });
     } catch(err) {
       console.error('Error in getAllReleaseNotes', err);
     }
   }
 
-  await Promise.all(Object.keys(matchingTags).map(async function(packageName, index) {
-    releaseNotes[packageName] = await Promise.all(matchingTags[packageName].map(async function(taggedRelease) {
-      let version = taggedRelease.name;
+  // matchingTags :: {<packageName>: [{name: <version>, commit, zipball_url, ...}, ...]}
+  await Promise.all(Object.keys(matchingTags).map(async function(packageName) {
+    // At this step keep changes grouped by the package name:
+    releaseNotes[packageName] = await Promise.all(matchingTags[packageName].map(async function(taggedRelease, index) {
+      // The 1st item is the prevVer that should be excluded (it is just used for the next one to get its prev):
+      if (index === 0) {
+        return;
+      }
+      const version = taggedRelease.name;
+      const currentReleaseSha = taggedRelease.commit.sha;
+      const previousRelease = index > 0 && matchingTags[packageName][index - 1].name;
```

Comment:
Might be worth extracting `index > 0 && matchingTags[packageName][index - 1]` out so you don't have to do this twice. It might also make this more readable.

Reply:
Fixed.